### DISCUSSION TITLE
Alguém usando .net Framework 4.5.x ainda?

### DISCUSSION THREAD

**Discussion by orochasamuel** [orochasamuel/fiscalbr-net#74] [Q&A]
Estou pensando em remover suporte aos frameworks antigos não suportados mais pela Microsoft, 4.5, 4.5.1 e 4.5.2. Ainda tem alguém utilizando em seus projetos essas versões "legado"?

![image](https://user-images.githubusercontent.com/15462690/218091954-59824e3f-3956-4d16-989d-8e45954f3fa0.png)
Fonte: https://learn.microsoft.com/pt-br/dotnet/framework/install/guide-for-developers

@marcosgerene @marcianobandeira @adrianotrentim @ebitencourt @ThalyaAutocom @PGama-Rodrigo @hbdbim @EderWillian @eduarda80

**Comment by hbdbim** [comment]
Estou no 4.8 migrando pra versão mais atual.

**Reply by marcianobandeira** [reply]
usando apenas atuais .net6 e .net7

**Comment by elieltonbitencourt** [comment]
estou na 4.7.2, migrando os projetos base para standard 2.0 para depois ir pra 7

**Comment by WashingtonAutocom** [comment]
Aqui usamos a 4.6.2.

**Comment by orochasamuel** [comment]
Recentemente desinstalei o VS 2019 e começou a acusar falta dos Developer Pack das versões 4.5.x.

Mantive suporte às versões 4.5.1 e 4.5.2 mas removi a 4.5.

A partir da versão 17.0.2 da EFD Fiscal e 6.0.2 da EFD Contribuições a lib não oferecerá mais suporte ao .net framework 4.5.

**Comment by candinho87** [comment]
A meu ver acho que deveria seguir as atualizações do dotnet(ex 8.0), ate porque esses frameworks ja foram descontinuados, maior parte dos usuarios apenas mantem para uso de reportviewer e afins, claro que entendo que tem muita gente ainda nesses frameworks, mas ja passou muitos anos, ja deu tempo mais que suficiente para migrar para net core, ate porque nao vai mudar muita coisa no codigo, e o proprio vs ja faz a migração de quase tudo, so revisar e mudar os relatorios de reportviewer para pdf e deu, no momento estou desenvolvendo uma app para comercio e usarei a biblioteca, porem irei migrar ela todinha para net 8

**Reply by adrianotrentim** [reply]
É que a coisa não funciona assim, eu por exemplo tenho um ERP "INTEIRO" com mais de 15 anos rodando pra industrias e engenharias em MVC 5 (e olha que mesmo com 15 anos, ainda falta coisa).... não dá pra simplesmente reescrever..... A Mega, que é uma gigante, ainda usa Desktop pra vc ter uma idéia.... demora muito e é muito caro pra reestruturar um software desse tamanho.... da ultima vez que levantamos aqui pra reescrever, chegamos em torno de 18 meses com 3 sêniors.... isso dá + de 1M (só por estimativa, que sempre fura).. Eu não sei o que vc tem em seu software, mas aqui por exemplo temos Multi-tenant por exemplo, que no framework nem existe, fizemos na unha nas entranhas do sistema, no .net core já existe, precisariamos rescrever toda essa parte, parte de validações nas entidades, serviços, enfim, estas coisas não são migradas automaticamente, tem que ser feitas na mão, com MUITA atenção.....

**Reply by candinho87** [reply]
Depende como vc estrutura o codigo, no meu caso foi super simples pois tudo era separado, e o app client é em wpf ainda, mas a api que foi migrada for muito simples, muito pouca coisa para ajustar, porem como temos por padrao 1 cliente(filiais inclusas)/servidor, nao tivemos problema algum com tenant

**Reply by marcosgerene** [reply]
@candinho87 permita-me discordar... 

Migrar um software legado de grande porte é sim muito duro. Se foi fácil é por algum desses motivos:

1. O projeto ja estava numa versão bem próxima da migrada
2. O projeto não é realmente grande
3. O projeto não atende clientes complexos.

Qualquer coisa além disso, quero um autógrafo seu.



@adrianotrentim 

Concordo com seu ponto, mas se o projeto tem todo esse tamanho e complexidade então para mante-lo, este deve ser rentável, logo, o projeto tem que considerar o custo de ser legado (caso contrário ele não vale a pena ser mantido...).




A discussão inicial era remover o suporte a .NET 4.5.x aqui. 
Acho no mínimo ruim manter esse suporte, mais uma vez, meu argumento baseia-se em quem paga a conta.

Tudo isso para dizer, o projeto aqui deve atender exclusivamente versões suportadas oficialmente pela MS.

**Comment by adrianotrentim** [comment]
@orochasamuel estamos em 4.8 aqui.
